backend/api/routes/files.py

```diff
@@ -151,50 +152,91 @@ def validate_project_quota(file_size: int, project: Project):
         )
 
 
-def s3_file_key(project_id: UUID, path: str) -> str:
+def s3_file_key(project_id: UUID, file_hash: str) -> str:
     """Generate s3 file key."""
-    return f"{project_id}-{path}"
+    to_be_hashed_str = f"{project_id}-{file_hash}-{constants.private_key}"
+    return hashlib.sha256(bytes(to_be_hashed_str, "utf-8")).hexdigest()
 
 
-def update_file_status_and_path(file: File, status: FileStatus, path: str):
+def update_file_status_and_path(file: File, status: str, path: str):
     """Update file's Status and Path."""
-    session = next(gen_session())
-    stmt = update(File).filter_by(id=file.id).values(status=status, path=path)
-    session.execute(stmt)
-    session.commit()
+    with DBSession.begin() as session:
+        stmt = update(File).filter_by(id=file.id).values(status=status, path=path)
+        session.execute(stmt)
+        session.commit()
+
+
+def update_file_status(file: File, status: str):
+    """Update file's Status."""
+    update_file_status_and_path(file, status, file.path)
+
+
+def update_file_path(file: File, path: str):
+    """Update file's path."""
+    update_file_status_and_path(file, file.status, path)
+
 
+def get_file_by_id(file_id: UUID) -> File:
+    """Get File instance by its id."""
+    with DBSession.begin() as session:
+        stmt = select(File).where(File.id == file_id)
+        file = session.execute(stmt).scalar()
+        if not file:
+            raise HTTPException(HTTPStatus.NOT_FOUND, f"File not found: {file_id}")
```

Comment:
This is not used in a FastAPI/HTTP context. Raising this error is thus irrelevant. Should be a `ValueError` I think